### PR TITLE
Don't re-scan previously visited blocks in helperexpansion.cpp

### PR DESCRIPTION
**Pull request by EgorBo**

### CODE REVIEW

**Comment by EgorBo** [comment]
@kunalspathak PTAL, that change I proposed for the helper expansion. Currently it is quite fragile and relies on a fact that expansion will put a call into a cold block or manually clear some flag. This PR skips all internal blocks added during expansion. Diffs: https://dev.azure.com/dnceng-public/public/_build/results?buildId=248685&view=results I expected to see some small TP improvement but I guess we don't expand helper calls that much.

File: src/coreclr/jit/helperexpansion.cpp [hunk 1]
```diff
@@ -93,8 +93,9 @@ PhaseStatus Compiler::fgExpandRuntimeLookups()
     return fgExpandHelper<&Compiler::fgExpandRuntimeLookupsForCall>(false);
 }
 
-bool Compiler::fgExpandRuntimeLookupsForCall(BasicBlock* block, Statement* stmt, GenTreeCall* call)
+bool Compiler::fgExpandRuntimeLookupsForCall(BasicBlock** pBlock, Statement* stmt, GenTreeCall* call)
```
Comment (kunalspathak) [inline src/coreclr/jit/helperexpansion.cpp:96]: Can you add method summary for this method?

File: src/coreclr/jit/helperexpansion.cpp [hunk 2]
```diff
@@ -435,9 +437,9 @@ PhaseStatus Compiler::fgExpandThreadLocalAccess()
 //                             that access fields marked with [ThreadLocal].
 //
 // Arguments:
-//    block - Block containing the helper call to expand
-//    stmt  - Statement containing the helper call
-//    call  - The helper call
+//    pBlock - Block containing the helper call to expand (may be updated in case of block splitting)
```
Comment (kunalspathak) [inline src/coreclr/jit/helperexpansion.cpp:440]:
```suggestion
//    pBlock - Block containing the helper call to expand. If expansion is performed, this is updated to the new block that was an outcome of block splitting.
```

Likewise at other places.

File: src/coreclr/jit/helperexpansion.cpp [hunk 3]
```diff
@@ -697,7 +701,7 @@ PhaseStatus Compiler::fgExpandHelper(bool skipRarelyRunBlocks)
         }
 
         // Expand and visit the last block again to find more candidates
-        while (fgExpandHelperForBlock<ExpansionFunction>(block))
+        while (fgExpandHelperForBlock<ExpansionFunction>(&block))
```
Comment (kunalspathak) [inline src/coreclr/jit/helperexpansion.cpp:704]: I would make sure that we indeed get an updated block if `fgExpandHelperForBlock()` returned `true.`

```c++
INDEBUG(BasicBlock* origBlock = block);
while (fgExpandHelperForBlock<ExpansionFunction>(&block))
{
   result = PhaseStatus::MODIFIED_EVERYTHING; 
#ifdef DEBUG
   assert(origBlock != block);
   origBlock = block;   
#endif
}
```

**Review by kunalspathak — CHANGES_REQUESTED** [review]

**Comment by EgorBo** [comment]
@kunalspathak thanks for the feedback, I believe I've addressed all comments

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by EgorBo** [comment]
failure is https://github.com/dotnet/runtime/issues/76454

**Pull request merged**